mathesar_ui/src/stores/table-data/columns.ts

```diff
@@ -113,6 +114,7 @@ export class ColumnsDataStore
       };
       this.set(storeData);
       this.fetchCallback?.(storeData);
+      await this.dispatch('columnsFetched', storeData.columns);
```

Comment:
It's fine to dispatch this event here, but in #1616, I've proposed that we remove the listener to the event. 

> I personally feel having this event system is super difficult to manage and reason about.

This is a pattern that @pavish designed. I agree that it's difficult to manage and reason about, and I have some other qualms about it too, which Pavish and I have debated on several occasions.

For _the particular use-case in this PR_, it doesn't seem like it's necessary to use this event system at all (as I've described in #1616).

But for other use-cases we do need _some system_ to keep our stores in sync when one change necessitates changes somewhere else. For example, if the user deletes a column, we need to make sure we remove references to it from our filtering/sorting/grouping stores. The pattern that I had proposed to Pavish was that we handle those "side effects" in a more manual manner. The calling context would be responsible for figuring out _what else_ needs to happen when we delete a column for example. That pattern I proposed is a little simpler and easier to reason about, but it's also more error-prone. Perhaps our UI has two wildly different mechanisms to delete a column. Then we'd need to make sure that each of them behaves in a similar manner, taking care to "clean up" after the column is deleted. I've come around to this `EventHandler` pattern, though it still doesn't feel optimal to me.

I'd be very curious to hear what sort of patterns you might prefer using here. This ties in to some of the conversation we had at our front end team meeting where I wondered out-loud whether a tool like [TanStack](https://tanstack.com/query/v4) might help us with some of these rough spots in our code. If I were to spend time trying to find a better pattern than our current `EventHandler` pattern, looking at TanStack (and similar libraries) would probably be the first direction I'd head. Since you've used React Query a bit, you might be able to offer some good insight here for us!